File: angular/ProductManagement/src/app/models/product.ts
export interface IProduct {
    productId: number;
    name: string;
    code: string;
    releaseDate: Date;
    description: string;
    price: number;
    starRating: number;
    imageUrl: string;
}

export class Product implements IProduct {
    constructor(public productId: number,
        public name: string,
        public code: string,
        public releaseDate: Date,
        public description: string,
        public price: number,
        public starRating: number,
        public imageUrl: string){}
}

File: .net core/DemoApplication/Demo.Contract/Models/Product.cs
namespace Demo.Contract.Models
{

    using System;
    using System.ComponentModel.DataAnnotations;

    public class Product
    {
        //[Required]
        public int ProductId { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public DateTime ReleaseDate { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public double StarRating { get; set; }
        public string ImageUrl { get; set; }
    }
}
